study1/week1/unique-binary-search-trees.java

```diff
@@ -0,0 +1,22 @@
+// https://leetcode.com/problems/unique-binary-search-trees/
+// References: https://st-lab.tistory.com/300
+class Solution {
+    // Time Complexity: O(n^2)
```

Comment:
Time complexity of this implementation will be exponential. I don't know the factor but at least `O(2^n)` it's because of the following.

 lets consider an example `n = 4`

`numTrees(4)` will call
- `numTrees(0)`, `numTrees(3)`
- `numTrees(1)`, `numTrees(2)`
- `numTrees(2)`, `numTrees(1)`
- `numTrees(3)`, `numTrees(0)`
you already need to call two `numTrees(3)` in `numTrees(4)`. meaning that every time you increase the number it will be at least two times more computation than the previous `numTrees(n-1)`, thus every time `n` increases by 1, the computation needed multiplies by two, thus this is exponential and at least `O(2^n)`.

You can make this `O(n)` by using [memoization](https://en.wikipedia.org/wiki/Memoization)